rviz_default_plugins/src/rviz_default_plugins/displays/depth_cloud/depth_cloud_display.cpp

```diff
@@ -0,0 +1,606 @@
+        sync_depth_color_->registerCallback(
+          std::bind(
+            &DepthCloudDisplay::processMessage, this,
+            std::placeholders::_1, std::placeholders::_2));
+
+        pointcloud_common_->color_transformer_property_->setValue("RGB8");
+      } else {
+        depthmap_tf_filter_->registerCallback(
+          std::bind(&DepthCloudDisplay::processDepthMessage, this, std::placeholders::_1));
+      }
+    }
+  } catch (const image_transport::TransportLoadException & e) {
+    setStatus(
+      rviz_common::properties::StatusProperty::Error, "Message",
+      QString("Error subscribing: ") + e.what());
+  }
+}
+
+void DepthCloudDisplay::caminfoCallback(sensor_msgs::msg::CameraInfo::ConstSharedPtr msg)
+{
+  std::lock_guard<std::mutex> lock(cam_info_mutex_);
+  cam_info_ = std::move(msg);
+}
+
+void DepthCloudDisplay::unsubscribe()
+{
+  clear();
+
+  sync_depth_color_.reset(new SynchronizerDepthColor(SyncPolicyDepthColor(queue_size_)));
+  depthmap_tf_filter_.reset();
+  depthmap_sub_.reset();
+  rgb_sub_.reset();
+  cam_info_sub_.reset();
+}
+
+void DepthCloudDisplay::clear()
+{
+  pointcloud_common_->reset();
+}
+
+void DepthCloudDisplay::update(float wall_dt, float ros_dt)
+{
+  pointcloud_common_->update(wall_dt, ros_dt);
+}
+
+void DepthCloudDisplay::reset()
+{
+  clear();
+  messages_received_ = 0;
+  setStatus(rviz_common::properties::StatusProperty::Ok, "Depth Map", "0 depth maps received");
+  setStatus(rviz_common::properties::StatusProperty::Ok, "Message", "Ok");
+}
+
+void DepthCloudDisplay::processDepthMessage(const sensor_msgs::msg::Image::ConstSharedPtr depth_msg)
+{
+  processMessage(depth_msg, sensor_msgs::msg::Image::ConstSharedPtr());
+}
+
+void DepthCloudDisplay::processMessage(
+  const sensor_msgs::msg::Image::ConstSharedPtr depth_msg,
+  const sensor_msgs::msg::Image::ConstSharedPtr rgb_msg)
+{
+  if (context_->getFrameManager()->getPause()) {
+    return;
+  }
+
+  std::ostringstream s;
+
+  ++messages_received_;
+  setStatus(
+    rviz_common::properties::StatusProperty::Ok, "Depth Map",
+    QString::number(messages_received_) + " depth maps received");
+  setStatus(rviz_common::properties::StatusProperty::Ok, "Message", "Ok");
+
+  sensor_msgs::msg::CameraInfo::ConstSharedPtr cam_info;
+  {
+    std::lock_guard<std::mutex> lock(cam_info_mutex_);
+    cam_info = cam_info_;
+  }
+
+  if (!cam_info || !depth_msg) {
+    return;
+  }
+
+  s.str("");
+  s << depth_msg->width << " x " << depth_msg->height;
+  setStatusStd(rviz_common::properties::StatusProperty::Ok, "Depth Image Size", s.str());
+
+  if (rgb_msg) {
+    s.str("");
+    s << rgb_msg->width << " x " << rgb_msg->height;
+    setStatusStd(rviz_common::properties::StatusProperty::Ok, "Image Size", s.str());
+
+    if (depth_msg->header.frame_id != rgb_msg->header.frame_id) {
+      std::stringstream errorMsg;
+      std::cerr << "Depth image frame id [" << depth_msg->header.frame_id.c_str()
```

Comment:
I think this wants to be:

```suggestion
      errorMsg << "Depth image frame id [" << depth_msg->header.frame_id.c_str()
```